### PR TITLE
CircleSort Algo in Kotlin

### LINKED ISSUE #2129: Circle Sort
- [x] C
- [x] C++
- [ ] C#
- [ ] CoffeeScript
- [ ] Dart
- [ ] Go - @aakashdinkar
- [x] Java
- [ ] JS - @HariniJeyaraman
- [x] Kotlin - @myselfanuj
- [x] Php - @mharshita
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by myselfanuj**
Fixes #2129

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Circle Sort Algo in Kotlin
- 

#### Languages Used:
-  Kotlin
- 

#### Files Added:
-  Circle_Sort.kts
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Circle_Sort/Circle_Sort.kts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    fun swap(array: Array<T>, i: Int, j: Int) {
+        val temp  = array[i]
+        array[i]  = array[j]
+        array[j]  = temp
+    }
+ 
+    var high  = hi
+    var low   = lo
+    val mid   = (hi - lo) / 2
+    var swaps = nSwaps
```
Comment (asha15) [inline Circle_Sort/Circle_Sort.kts]: you kept two spaces before the equal, please correct it

File: Circle_Sort/Circle_Sort.kts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    while (low < high) {
+        if (array[low] > array[high]) {
+            swap(array, low, high)
+            swaps++
+        }
+        low++
+        high--
+    }
```
Comment (asha15) [inline Circle_Sort/Circle_Sort.kts]: keep a line space after this

File: Circle_Sort/Circle_Sort.kts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    if (low == high)
+        if (array[low] > array[high + 1]) {
+            swap(array, low, high + 1)
+            swaps++
+        }
```
Comment (asha15) [inline Circle_Sort/Circle_Sort.kts]: keep a line space after this

File: Circle_Sort/Circle_Sort.kts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+fun main(args: Array<String>) {
+    val array = arrayOf(6, 7, 8, 9, 2, 5, 3, 4, 1)
+    println("Original: ${array.asList()}")
+    while (circleSort(array, 0, array.size - 1, 0) != 0) ; // empty statement
+    println("Sorted  : ${array.asList()}")
+    println()
+    val array2 = arrayOf("the", "quick", "brown", "fox", "jumps", "over", "the", "lazy", "dog")
+    println("Original: ${array2.asList()}")
+    while (circleSort(array2, 0, array2.size - 1, 0) != 0) ;
+    println("Sorted  : ${array2.asList()}")
```
Comment (asha15) [inline Circle_Sort/Circle_Sort.kts]: your code looks so compact, keep some line spaces

**Comment by myselfanuj** [comment]
I did changes in file please review it. @asha15

**Comment by myselfanuj** [comment]
Please review my comment its almost a week but you didn't reviewed my PR @asha15 and @jainaman224

**Comment by myselfanuj** [comment]
Please review my PR I did all necessary changes in the file. @asha15 and @jainaman224.

**Comment by asha15** [comment]
Sorry for the delay. I had a trouble with internet connection

**Comment by muskankhedia** [comment]
Is this PR has to be calculated for GSSoC or not as no labels have been added to it?

**Pull request merged**